File: College_shared_code/DSA/BinarySearch.c
#include <stdio.h>

int main()
{
    int first = 0, last, middle, size, i, SearchElem, list[100];

    printf("Enter the number of elements : ");
    scanf("%d",&size);

    printf("Enter %d integer elements in Assending order : \n", size);
    for (i = 0; i < size; i++)
    {
        scanf("%d",&list[i]);
    }

    printf("Enter value to be searched: ");
    scanf("%d", &SearchElem);

    last = size - 1;
    middle = (first+last)/2;

    while (first <= last) 
    {
        if (list[middle] < SearchElem)
        {
            first = middle + 1;    
        }
        else if (list[middle] == SearchElem) 
        {
            printf("Element found at index %d.\n",middle);
            break;
        }
        else
        {
            last = middle - 1;
        }

        middle = (first + last)/2;
    }

    if (first > last)
    {
        printf("Element Not found in the list.");
    }

    return 0;
}

File: College_shared_code/PythonPracticalPrograms/BinarySearch.py
# A program to implement Binary search

# declare an empty list
arr = []

# variable to store number of elements
n = int(input("Enter the number of elements to enter : "))

# taking input from the user
print("Enter the elements one by one.")
for i in range(n):
    a = int(input(f"Element [{i}] : "))
    arr.append(a)

# sort the list
arr.sort()

# displaying the elements in the list
print("The elements in the list are : ")
print(arr)

# variable to store the element to be searched
search = int(input("Enter the element to search : "))

# variables required to perform Binary Search
first = 0
last = n

# While loop to initiate the search
while (first < last):
    mid = int((first + last) // 2)

    if (arr[mid] == search):
        print(f"Element found at index {mid}.")
        break
        
    elif (search < arr[mid]):
        last = mid - 1

    else:
        first = mid + 1

# If element not found display appropriate message
if (first >= last):
    print("Element not found in the list.")
